packages/react-table/src/useTable/index.ts

```diff
@@ -6,31 +6,57 @@ import {
     LogicalFilter,
     useTable as useTableCore,
     useTableProps as useTablePropsCore,
-    useTableReturnType,
+    useTableReturnType as useTableReturnTypeCore,
+    useTableNoPaginationReturnType as useTableNoPaginationReturnTypeCore,
 } from "@pankod/refine-core";
 import { useTable as useTableRT, PluginHook, TableOptions } from "react-table";
 
 export type UseTableReturnType<TData extends BaseRecord = BaseRecord> =
     ReturnType<typeof useTableRT> & {
-        refineCore: useTableReturnType<TData>;
+        refineCore: useTableReturnTypeCore<TData>;
     };
 
+export type UseTableNoPaginationReturnType<
+    TData extends BaseRecord = BaseRecord,
+> = ReturnType<typeof useTableRT> & {
+    refineCore: useTableNoPaginationReturnTypeCore<TData>;
+};
+
 export type UseTableProps<
     TData extends BaseRecord = BaseRecord,
     TError extends HttpError = HttpError,
 > = {
     refineCoreProps?: useTablePropsCore<TData, TError>;
 } & TableOptions<{}>;
 
-export const useTable = <
+export function useTable<
     TData extends BaseRecord = BaseRecord,
     TError extends HttpError = HttpError,
 >(
-    { refineCoreProps, ...rest }: UseTableProps<TData, TError>,
+    props: UseTableProps<TData, TError> & { hasPagination?: true },
```

Comment:
```suggestion
    props: UseTableProps<TData, TError> & { hasPagination?: boolean },
```
should it be like this?

Reply:
This is actually done for overloading the function; `useTable` returns different values by the `hasPagination` value. For `current`, `pageSize` etc. we return `undefined` if pagination is disabled, otherwise they return `number` and `React.Dispatch`